subsys/net/ip/icmp.c

```diff
@@ -0,0 +1,589 @@
+/** @file
+ * @brief ICMP related functions
+ */
+
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Use highest log level if both IPv4 and IPv6 are defined */
+#if defined(CONFIG_NET_IPV4) && defined(CONFIG_NET_IPV6)
+#define ICMP_LOG_LEVEL COND_CODE_1(CONFIG_NET_ICMPV4_LOG_LEVEL > CONFIG_NET_ICMPV6_LOG_LEVEL, \
+	(CONFIG_NET_ICMPV4_LOG_LEVEL), (CONFIG_NET_ICMPV6_LOG_LEVEL))
```

Comment:
This doesn't appear to work after some testing, an additional `#if` will be required:

```c
#if defined(CONFIG_NET_IPV4) && defined(CONFIG_NET_IPV6)

#if CONFIG_NET_ICMPV4_LOG_LEVEL > CONFIG_NET_ICMPV6_LOG_LEVEL
#define ICMP_LOG_LEVEL CONFIG_NET_ICMPV4_LOG_LEVEL
#else
#define ICMP_LOG_LEVEL CONFIG_NET_ICMPV6_LOG_LEVEL
#endif

#elif defined(CONFIG_NET_IPV4)
#define ICMP_LOG_LEVEL CONFIG_NET_ICMPV4_LOG_LEVEL
#elif defined(CONFIG_NET_IPV6)
#define ICMP_LOG_LEVEL CONFIG_NET_ICMPV6_LOG_LEVEL
#else
#define ICMP_LOG_LEVEL LOG_LEVEL_INF
#endif
```

Reply:
It looked very cool, well back to normal checks then :(